### PR TITLE
Models

### PR DESCRIPTION
**Pull request by charlesgobina**
For this milestone, I
* Created User, Post, Like and Comment models and implemented the specified methods

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class User < ApplicationRecord
+  has_many :post, class_name: 'Post', foreign_key: 'post_id'
+  has_many :comment, class_name: 'Comment', foreign_key: 'author_id', dependent: :destroy
+  has_many :like, class_name: 'Like', foreign_key: 'author_id', dependent: :destroy
```
Comment (rloterh) [inline app/models/user.rb]: - [ ] As per the given _ERD_, the _User_ **association** should have _author_id_ referenced as **foreign key** in _Post_, _Comment,_ and _Like_ models. Kindly ensure to improve this aspect, as can be seen below:
```ruby
has_many :posts, foreign_key: 'author_id'
has_many :comments, foreign_key: 'author_id'
has_many :likes, foreign_key: 'author_id'
```

File: app/models/post.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+  has_many :comment, class_name: 'Comment', foreign_key: 'author_id', dependent: :destroy
+  has_many :like, class_name: 'Like', foreign_key: 'author_id', dependent: :destroy
```
Comment (rloterh) [inline app/models/post.rb]: - [ ] You have defined **unrequired foreign keys**. Kindly note that foreign keys associations between the _User_ and the other models (_Comment_, _Post_, and _Like_) are already defined in the _User_ model. Associations defined in the Post model should simply look like:
```ruby
# app/models/post.rb
belongs_to :author, class_name: 'User'
has_many :comments
has_many :likes
  
...
```

File: app/models/post.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+  after_save :update_post_counter
+
+  def update_post_counter
```
Comment (rloterh) [inline app/models/post.rb:14]: - [ ] To ensure that the `update_post_counter` is properly accessed kindly mark it as _private_

File: app/models/user.rb [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+
```
Comment (rloterh) [inline app/models/user.rb:5]: - [ ] Current_user method is missing in _app/controllers/application_controller.rb_. Kindly ensure to define the same in  _app/controllers/application_controller.rb_ smilar to:
```ruby
# app/controllers/application_controller.rb
...

def current_user
  User.first
end
```

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Charles, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 🚦 🚦 🚦  

## Good Points 👍  
- Descriptive Pull Request
- No linter errors


## Required Changes ♻️
_Kindly check the comments under the review._

<br>
Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/user.rb (hunk 4 above)
Comment (charlesgobina) [reply app/models/user.rb:5]: Hi @rloterh

Thank you for the feedback. I just want to clarify a doubt. Please what's the use of the current_user method. 

Thank you

**Review by BrianSammit — APPROVED** [review]
# STATUS: APPROVED :green_circle: :clap: 

Hello @charlesgobina, 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

TO HIGHLIGHT:

- Your project is professional 
- Your repo is professional 
- Well structured files 


Now you can merge this branch and move on :1st_place_medal: 
HAPPY CODING AND GOOD LUCK WITH THE COMMING ONES :crossed_fingers: :four_leaf_clover: 

well-1

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/user.rb (hunk 4 above)
Comment (rloterh) [reply app/models/user.rb:5]: @charlesgobina 
Just as the name is, the current user points to the particular user who is presently logged in to the said application at a said point in time

**Pull request merged**